src/Portal.sol

```diff
@@ -157,6 +390,13 @@ abstract contract Portal is NttManagerNoRateLimiting, IPortal {
      */
     function _mintOrUnlock(address recipient_, uint256 amount_, uint128 index_) internal virtual {}
 
+    /**
+     * @dev   HubPortal:   locks amount_` M tokens.
+     *        SpokePortal: burns `amount_` M tokens.
+     * @param amount_ The amount of M tokens to lock/burn.
+     */
+    function _burnOrLock(uint256 amount_) internal virtual {}
```

Comment:
I don't see this function being overriden in HubPortal.
I suppose it's because it performs no operation in this case.
Maybe it should still be overriden in HubPortal and we could add a comment specifying that no operation is being performed since M token has already been sent to the contract.